durabletask/task.py

```diff
@@ -313,6 +296,92 @@ def get_completed_tasks(self) -> int:
         return self._completed_tasks
 
 
+class CompletableTask(Task[T]):
+
+    def __init__(self):
+        super().__init__()
+        self._retryable_parent = None
+
+    def complete(self, result: T):
+        if self._is_complete:
+            raise ValueError('The task has already completed.')
+        self._result = result
+        self._is_complete = True
+        if self._parent is not None:
+            self._parent.on_child_completed(self)
+
+    def fail(self, message: str, details: pb.TaskFailureDetails):
+        if self._is_complete:
+            raise ValueError('The task has already completed.')
+        self._exception = TaskFailedError(message, details)
+        self._is_complete = True
+        if self._parent is not None:
+            self._parent.on_child_completed(self)
+
+
+class RetryableTask(CompletableTask[T]):
+
+    def __init__(self, retry_policy: RetryPolicy, action: pb.OrchestratorAction,
+                 is_sub_orch: bool) -> None:
+        super().__init__()
+        self._action = action
+        self._retry_policy = retry_policy
+        self._retry_count = 0
+        self._first_attempt: datetime
+        self._is_sub_orch = is_sub_orch
+
+    def try_completion(self) -> bool:
+        if self._retry_count >= self._retry_policy.max_number_of_attempts - 1:
+            return True
+        else:
+            return False
+    
+    def set_first_attempt(self, first_attempt: datetime) -> None:
+        self._first_attempt = first_attempt
+
+    def increment_retry_count(self) -> None:
+        self._retry_count += 1
+    
+    def compute_next_delay(self) -> timedelta:
+        next_delay = timedelta.min
```

Comment:
Here you're assigning `next_delay` to `timedelta.min` but aren't referencing it nor are you reassigning it. Seems like this can be removed.

Reply:
Removed it